dbms/src/Parsers/ASTSelectQuery.cpp

```diff
@@ -216,11 +216,11 @@ static const ASTArrayJoin * getFirstArrayJoin(const ASTSelectQuery & select)
 static const ASTTablesInSelectQueryElement * getFirstTableJoin(const ASTSelectQuery & select)
 {
     if (!select.tables)
-        return {};
+        return nullptr;
```

Comment:
Do we really need it? I'd prefere empty list to be able to change pointer to object without need to change return statements all over the code.